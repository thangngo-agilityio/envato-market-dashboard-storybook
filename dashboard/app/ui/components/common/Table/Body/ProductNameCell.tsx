import { Flex, Td, Text, Tooltip } from '@chakra-ui/react';
import { memo } from 'react';

// Types
import { TDataSource } from '@/lib/interfaces';

const ProductNameCellComponent = ({ name }: TDataSource): JSX.Element => (
  <Td
    py={5}
    pr={5}
    pl={0}
    fontSize="md"
    color="text.primary"
    fontWeight="semibold"
    textAlign="left"
    w={{ base: 200, xl: 220, '3xl': 200, '6xl': 250 }}
  >
    <Flex alignItems="center" gap="10px">
      <Tooltip
        minW="max-content"
        placement="bottom-start"
        label={name as string}
      >
        <Text
          display="block"
          fontSize="md"
          fontWeight="semibold"
          wordBreak="break-all"
          textOverflow="ellipsis"
          overflow="hidden"
          pr={10}
          flex={1}
          w={{ base: 200, xl: 220, '3xl': 200, '6xl': 250 }}
        >
          {name as string}
        </Text>
      </Tooltip>
    </Flex>
  </Td>
);

const ProductNameCell = memo(ProductNameCellComponent);

export default ProductNameCell;
