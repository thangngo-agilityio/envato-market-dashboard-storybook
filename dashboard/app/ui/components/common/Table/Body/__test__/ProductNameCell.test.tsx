import { render } from '@testing-library/react';
import { Table } from '@chakra-ui/react';

// Components

import ProductNameCell from '../ProductNameCell';

const setup = () =>
  render(<ProductNameCell id={1} name="Xbox one" />, {
    wrapper: Table,
  });

describe('ProductNameCell', () => {
  it('Match to snapshot', () => {
    const { container } = setup();

    expect(container).toMatchSnapshot();
  });

  it('Render', () => {
    const { getByText } = setup();

    expect(getByText('Xbox one')).toBeDefined();
  });
});
