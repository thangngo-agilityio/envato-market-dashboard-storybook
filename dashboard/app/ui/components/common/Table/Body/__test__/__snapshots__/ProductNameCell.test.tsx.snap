// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductNameCell Match to snapshot 1`] = `
<div>
  <table
    class="chakra-table css-0"
  >
    <td
      class="css-xpdvke"
    >
      <div
        class="css-1h2sv0j"
      >
        <p
          class="chakra-text css-s51d82"
        >
          Xbox one
        </p>
      </div>
    </td>
  </table>
</div>
`;
